azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebAppDeploymentSlot.java

```diff
@@ -0,0 +1,201 @@
+        return azureAppService.webapp(entity.getResourceGroup(), entity.getWebappName());
+    }
+
+    @Override
+    public IWebAppDeploymentSlotCreator create() {
+        return new WebAppDeploymentSlotCreator();
+    }
+
+    @Override
+    public void start() {
+        getDeploymentSlotClient().start();
+    }
+
+    @Override
+    public void stop() {
+        getDeploymentSlotClient().stop();
+    }
+
+    @Override
+    public void restart() {
+        getDeploymentSlotClient().restart();
+    }
+
+    @Override
+    public void delete() {
+        getDeploymentSlotClient().parent().deploymentSlots().deleteByName(slotEntity.getName());
+    }
+
+    @Override
+    public void deploy(File file) {
+        deploy(AppServiceUtils.getDeployTypeByFileExtension(file), file);
+    }
+
+    @Override
+    public void deploy(DeployType deployType, File file) {
+        getDeploymentSlotClient().deploy(com.azure.resourcemanager.appservice.models.DeployType.fromString(deployType.getValue()), file);
+    }
+
+    @Override
+    public boolean exists() {
+        return getDeploymentSlotClient(true) != null;
+    }
+
+    @Override
+    public String hostName() {
+        return getDeploymentSlotClient().defaultHostname();
+    }
+
+    @Override
+    public String state() {
+        return getDeploymentSlotClient().state();
+    }
+
+    @Override
+    public Runtime getRuntime() {
+        return AppServiceUtils.getRuntimeFromWebApp(getDeploymentSlotClient());
+    }
+
+    @Override
+    public PublishingProfile getPublishingProfile() {
+        return AppServiceUtils.getPublishingProfile(getDeploymentSlotClient().getPublishingProfile());
+    }
+
+    @Override
+    public WebAppDeploymentSlotEntity entity() {
+        return slotEntity;
+    }
+
+    private com.azure.resourcemanager.appservice.models.DeploymentSlot getDeploymentSlotClient() {
+        return getDeploymentSlotClient(false);
+    }
+
+    private synchronized com.azure.resourcemanager.appservice.models.DeploymentSlot getDeploymentSlotClient(boolean force) {
+        if (deploymentSlotClient == null || force) {
+            try {
+                final WebApp webAppService = StringUtils.isNotEmpty(slotEntity.getId()) ?
+                        azureClient.webApps().getById(slotEntity.getId().substring(0, slotEntity.getId().indexOf("/slots"))) :
+                        azureClient.webApps().getByResourceGroup(slotEntity.getResourceGroup(), slotEntity.getWebappName());
+                deploymentSlotClient = StringUtils.isNotEmpty(slotEntity.getId()) ? webAppService.deploymentSlots().getById(slotEntity.getId()) :
+                        webAppService.deploymentSlots().getByName(slotEntity.getName());
+                slotEntity = AppServiceUtils.getWebAppDeploymentSlotEntity(deploymentSlotClient);
+            } catch (Exception e) {
+                // SDK will throw exception when resource not founded
+                return null;
+            }
```

Comment:
we do nothing for this exception ?

Reply:
Yes, we expect null for get a non-exists resource instead of exception